representation_errors/RE.py

```diff
@@ -0,0 +1,168 @@
+# (C) Crown Copyright, Met Office. All rights reserved.
+#
+# This file is part of ocean_error_covs and is released under the BSD 3-Clause license.
+# See LICENSE in the root of the repository for full licensing details.
+################## Python packages ###########################
+import os
+import numpy as np
+################## Code modules ##############################
+import modules.arrays as arrays
+from modules.io_data import IO
+from modules.utils import Utils
+from modules.masks import applyMask
+from representation_errors.calcREs import CalcREs
+
+# Initialising the classes
+IO = IO()
+Utils = Utils()
+applyMask = applyMask()
+CalcREs = CalcREs()
+
+def RE_unresolved_scales(list_of_fdbackfiles, model_mesh_file, obs_type="SLA",
+                 outfilename="RE_stats.nc", source_types=[], qc_val=[], min_num_obs=5,
+                 depth_lev=[0], latcen='gphit', loncen='glamt', latcor='gphit',
+                 loncor='glamf', mask='tmaskutil', thinning=1, lon_discontinuity=True):
+
+    """ Top-level routine that calculates the representation error due to
+        model unresolved scales based on Oke & Sakov (2008).
+        It calculates the average of all observations falling within
+        a grid cell and then computes their standard deviation. These
+        stats are then summed and stored.
+
+    *************** PARAMETERS *******************
+    1. list_of_fdbackfiles: list of feedback files
+    2. model_mesh_file: model mesh filename
+    3. obs_type: observation type to process
+    4. outfilename: name of the output file
+    5. source_types: observation id types to process
+                     (default: [], which means process all)
+    6. qc_val: QC values to filter obs
+                     (default: [], which means do not filter any obs)
+    7. min_num_obs: minimum number of obs to proceed with calculations
+    8. depth: list with depth levels (default is [0] - only surface)
+    9. latcen: variable name of the latitudes at the center of grid cell
+    10. loncen: variable name of the longitudes at the center of grid cell
+    11. latcor: variable name of the latitudes at the corners of grid cell
+    12. loncor: variable name of the longitudes at the corners of grid cell
+    13. mask: mask array (None in case mask is not provided)
+    14. thinning: number to subsample obs
+    15. lon_discontinuity: True in case there is discontinuity in longitude
+                           (e.g. 180 to -180)
+    """
+    # Check list of feedack files
+    list_of_fdbackfiles = Utils.check_files(list_of_fdbackfiles)
+    if list_of_fdbackfiles is None:
+       raise ValueError("[ERROR] FEEDBACK FILES NOT FOUND")
+
+    # Check model mesh file
+    model_mesh_file = Utils.check_files([model_mesh_file])
+    if list_of_fdbackfiles is None:
```

Comment:
This should be "if model_mesh_file is None:"

Reply:
Thanks for spotting this bug.